### PR TITLE
Migrate splink_vis_utils.js changes to upstream repo

### PR DESCRIPTION
**Pull request by RobinL**
This PR doesn't add any new functionality, I have just moved the various fixes/improvements to splink_vis_utils.js (which powers the comparison viewer and cluter studio dashboards) upstream to the source repo at https://github.com/robinL/splink_vis_utils.

See the readme there for more info about how to develop these tools further in future

This PR corresponds to [this commit](https://github.com/RobinL/splink_vis_utils/pull/6/commits/ceaaa0b3eb6b774549c1321fc6018719344ecf01)

The list of changes that needs migrating is:

- [x] https://github.com/moj-analytical-services/splink/pull/814
- [x] https://github.com/moj-analytical-services/splink/pull/1016
- [x] https://github.com/moj-analytical-services/splink/pull/1462
- [x] https://github.com/moj-analytical-services/splink/pull/1470
- [x] https://github.com/moj-analytical-services/splink/pull/1596
- [x] https://github.com/moj-analytical-services/splink/pull/1463

I've moved all the licences to `splink/files/DEPENDENCY_LICENSES.txt` so they don't have to be included in the html, and checked the file to ensure that they're all permissive/appropriate for us to include

### CODE REVIEW

**Comment by RobinL** [comment]
Having real problems with this.

The source of the problems seems to be https://github.com/observablehq/runtime/releases/tag/v5.0.0

This has changed the format of the javascript package downloaded like 
npm install https://api.observablehq.com/d/c43e4e0aeb54d263@774.tgz\?v\=3

A specific problem here is that it simple does not run at all offline - it tries to load in e.g. d3 and fails.

Another problem is when you're using observablehq, you are forced to use the 5.0.0 runtime, so there's no way of editing within observable under the 4.x runtime. So there's little point in finding a workaround.

I will need to find a way of running everything without the standard library, which is possible, see here:
https://github.com/observablehq/runtime#Runtime

File: splink/cluster_studio.py [hunk 1]
```diff
@@ -252,6 +252,7 @@ def render_splink_cluster_studio_html(
 
     files = {
         "embed": "files/external_js/vega-embed@6.20.2",
+        "stdlib": "files/external_js/stdlib.js@5.8.3",
```
Comment (RobinL) [inline splink/cluster_studio.py:255]: I was unable to find a way of using stdlib without inlineing it, hence the inclusion

File: splink/files/splink_cluster_studio/cluster_template.j2 [hunk 2]
```diff
@@ -18,13 +18,38 @@
...
+{# And then globally load the functions we need #}
+{# This also seems to avoid the problem of stdlib trying and failing to load d3 from a CDNs #}
+{# and causing the whole notebook to error #}
 
 
-    <script>{{svu_text}}</script>
+<script>
+const lib = new window.observablehq.Library()
+Generators = lib.Generators
+width = lib.width()
```
Comment (RobinL) [inline splink/files/splink_cluster_studio/cluster_template.j2:44]: we then grab the functions we need here.  i tried making them splink_vis_utils dependencies, but they're not exported from stdlib in the normal way, so it wasn't possible

**Review by ADBond — APPROVED** [review]
Have had a play around and this all looks good to me 👍 
Haven't experimented with the upstream repo yet, but this all seems to be working as expected.

**Comment by RobinL** [comment]
Super thank you

**Pull request merged**